test/common/http/http2/codec_impl_test.cc

```diff
@@ -340,7 +348,7 @@ TEST_P(Http2CodecImplTest, InvalidContinueWithFin) {
   request_encoder_->encodeHeaders(request_headers, true);
 
   TestResponseHeaderMapImpl continue_headers{{":status", "100"}};
-  EXPECT_THROW(response_encoder_->encodeHeaders(continue_headers, true), CodecProtocolException);
+  EXPECT_THROW(response_encoder_->encodeHeaders(continue_headers, true), ClientCodecError);
```

Comment:
question about this just generally -- aren't these non-dispatching throws? Eventually will not be statuses to errors, but will be ENVOY_BUGs?

Reply:
These exceptions are thrown only by and though the test harness (i.e. they do not go through codec under test). They thrown when the `dispatch` method of the codec under test returns an error. These throws will never be affected by the changes to codec, they are specific to this test suite only.

Reply:
Hmm. I see. Yeah, CodecClientError is test specific. 
Hmm... I think I got it. This is the non-dispatching errors I was concerned about testing
https://github.com/envoyproxy/envoy/blob/8cb2958f29af3e45e87653356b6099bd3a4cbfbf/source/common/http/http2/codec_impl.cc#L927

But you're right, it gets caught as a test-only throw right now. Trying to wrap my head around how it will be handled.